### PR TITLE
Taxonomy:  Check for empty term after DB sanitization in wp_insert_term

### PR DESCRIPTION
**Pull request by fgiannar**
When inserting a new term in the database, `wp_insert_term` will check if the term is empty and return a corresponding error.
Afterwards the term is sanitized and inserted in the database. However, there's a chance the term is empty after the DB sanitization.
This is never checked though and will result in 1) empty terms in the DB and 2) possible OOM issues in `wp_insert_term` when checking if the term name matches an existing one because of using `get_terms` with an empty name argument

The current PR:
- Adds a check for empty term name after the term is sanitized and returns an error if the term is empty.
- Adds a corresponding unit test

Trac ticket: https://core.trac.wordpress.org/ticket/59995

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: tests/phpunit/tests/term/wpInsertTerm.php [hunk 1]
```diff
@@ -895,6 +895,16 @@ public function test_wp_insert_term_with_null_description() {
...
 	}
 
+	public function test_wp_insert_term_with_empty_name_after_db_sanitization() {
+		$term = wp_insert_term(
+			'<script>Hello</script>',
+			'post_tag'
+		);
+
+		$this->assertWPError( $term );
+		$this->assertSame( 'invalid_term_name', $found->get_error_code() );
```
Comment (fgiannar) [inline tests/phpunit/tests/term/wpInsertTerm.php]:
```suggestion
		$this->assertSame( 'invalid_term_name', $term->get_error_code() );
```

**Review by fgiannar — COMMENTED** [review]
Typo

File: tests/phpunit/tests/term/wpInsertTerm.php [hunk 2]
```diff
@@ -895,6 +895,16 @@ public function test_wp_insert_term_with_null_description() {
 		$this->assertSame( '', $term_object->description );
 	}
 
+	public function test_wp_insert_term_with_empty_name_after_db_sanitization() {
+		$term = wp_insert_term(
+			'<script>Hello</script>',
```
Comment (fgiannar) [inline tests/phpunit/tests/term/wpInsertTerm.php]:
```suggestion
			'<script>alert("hello")</script>',
```

File: tests/phpunit/tests/term/wpInsertTerm.php [hunk 3]
```diff
@@ -895,6 +895,16 @@ public function test_wp_insert_term_with_null_description() {
 		$this->assertSame( '', $term_object->description );
 	}
 
+	public function test_wp_insert_term_with_empty_name_after_db_sanitization() {
+		$term = wp_insert_term(
+			'<script>alert("hello")</script>',
```
Comment (fgiannar) [inline tests/phpunit/tests/term/wpInsertTerm.php]:
```suggestion
			'<script>onclick=alert("hello")</script>',
```

File: src/wp-includes/taxonomy.php [hunk 4]
```diff
@@ -2434,6 +2434,10 @@ function wp_insert_term( $term, $taxonomy, $args = array() ) {
 	$description = wp_unslash( $args['description'] );
 	$parent      = (int) $args['parent'];
 
+	if ( '' === $name ) {
```
Comment (kraftbj) [inline src/wp-includes/taxonomy.php:2438]:
```suggestion
	// Santiziation could clean the name to an empty string that must be checked again. See https://core.trac.wordpress.org/ticket/59995
	if ( '' === $name ) {
```

**Review by kraftbj — APPROVED** [review]
This tests well for me with a dirty term name. Hotfixing in a production environment also resulted in expected change without any known regressions at this time. Unit test appears sufficient along side existing tests checking valid term names.

File: src/wp-includes/taxonomy.php [hunk 5]
```diff
@@ -2434,6 +2434,11 @@ function wp_insert_term( $term, $taxonomy, $args = array() ) {
 	$description = wp_unslash( $args['description'] );
 	$parent      = (int) $args['parent'];
 
+	// Santiziation could clean the name to an empty string that must be checked again. See https://core.trac.wordpress.org/ticket/59995
```
Comment (fgiannar) [inline src/wp-includes/taxonomy.php]:
```suggestion
	// Sanitization could clean the name to an empty string that must be checked again. See https://core.trac.wordpress.org/ticket/59995
```

**Comment by SergeyBiryukov** [comment]
Thanks for the PR! Merged in [r57251](https://core.trac.wordpress.org/changeset/57251).

**Pull request closed without merging**